src/adapters/MorphoMarketV1Adapter.sol

```diff
@@ -130,16 +132,9 @@ contract MorphoMarketV1Adapter is IMorphoMarketV1Adapter {
             MarketParams({loanToken: loanToken, collateralToken: collateralToken, oracle: oracle, irm: irm, lltv: lltv});
     }
 
-    function realAssets() external view returns (uint256) {
-        return allocation() != 0 ? expectedSupplyAssets() : 0;
-    }
-
-    /* INTERNAL FUNCTIONS */
-
-    function expectedSupplyAssets() internal view returns (uint256) {
+    function realAssets() public view returns (uint256) {
```

Comment:
If we completely remove the `allocation == 0` check, then a market with leftover shares but that is completely deallocated could later be called. So an emergency deallocation would not completely prevent the vault from calling the market.

The current MarketV1 adapter [prevents this](https://github.com/morpho-org/vault-v2/blob/main/src/adapters/MorphoMarketV1Adapter.sol#L113), and the current VaultV1 adapter [prevents this as well](https://github.com/morpho-org/vault-v2/blob/main/src/adapters/MorphoVaultV1Adapter.sol#L111).

Reply:
why don't we want to avoid calling the market? if it reverts, then you would do a removeAdapter, right?